lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java

```diff
@@ -103,14 +122,34 @@ public Iterator<Document> processDocument(Document doc) throws StageException {
       parseInputStream(doc, inputStream);
 
     } else if (doc.has(filePathField)) {
-
+      // get fileObject from path
       String filePath = doc.getString(filePathField);
+      FileObject file = FileUtils.getFileObject(filePath, fsManager);
+
+      // if file null, don't process document
+      if (file == null) {
+        return null;
+      }
+
+      // getting inputStream and catching errors gracefully
+      InputStream inputStream;
+      try {
+        // file.getContent() never returns null
+        // https://commons.apache.org/proper/commons-vfs/commons-vfs2/apidocs/org/apache/commons/vfs2/FileObject.html#getContent--
+        inputStream = file.getContent().getInputStream();
+      } catch(FileSystemException e) {
+        log.warn("Error getting inputStream from file at: {}", filePath, e);
+        return null;
+      }
+
+      parseInputStream(doc, inputStream);
 
+      // close file after, do not need to check if file is null as we have already checked above
       try {
-        InputStream inputStream = FileUtils.getInputStream(filePath);
-        parseInputStream(doc, inputStream);
-      } catch (IOException e) {
-        throw new StageException("InputStream cannot be parsed or created", e);
+        // closes the file, and its content, including any open stream
+        file.close();
```

Comment:
If an exception is thrown. file won't close. Better to wrap the creation of FileObject file in a try-with-resources and then we don't have to call close at all and we know that try will close the resource no matter what happens e.g.

try (FileObject file = FileUtils.getFileObject(filePath, fsManager)) {
 ...
}